pkg/receive/handler.go

```diff
@@ -294,6 +307,24 @@ func (h *Handler) TransformAndValidateWriteRequest(next http.Handler, labels ...
 			return
 		}
 
+		for _, ts := range wreq.GetTimeseries() {
```

Comment:
We could avoid entering these nested loops when no label value needs to be enforced.

Reply:
Good catch. Thanks